tests/python_package_test/test_dask.py

```diff
@@ -263,13 +273,17 @@ def test_classifier(output, task, client):
     p2_proba = local_classifier.predict_proba(X)
     s2 = local_classifier.score(X, y)
 
-    assert_eq(s1, s2)
-    assert_eq(p1, p2)
-    assert_eq(y, p1)
-    assert_eq(y, p2)
-    assert_eq(p1_proba, p2_proba, atol=0.01)
-    assert_eq(p1_local, p2)
-    assert_eq(y, p1_local)
+    if boosting_type == 'rf' and output == 'dataframe-with-categorical':
+        # https://github.com/microsoft/LightGBM/issues/4118
+        assert_eq(s1, s2, atol=0.01)
+        assert_eq(p1_proba, p2_proba, atol=0.8)
+    else:
+        assert_eq(p1, p2)
```

Comment:
Why are we dropping `assert_eq(s1, s2)` from the `else` branch?

Reply:
That was totally my bad, I lost it when splitting the statements. I've included it in 
5d7001d

Reply:
thanks for catching it @StrikerRUS ! I missed it too